reduction_scripts/reduce_data.py

```diff
@@ -0,0 +1,848 @@
+#! /usr/bin/env python
+
+import sys
+import os
+import pickle
+from astropy.io import fits as pyfits
+import pywifes
+import gc
+import datetime
+import numpy as np
+from data_classifier import classify
+import json
+
+#------------------------------------------------------------------------
+start_time = datetime.datetime.now()
+#------------------------------------------------------------------------
+
+# Read the raw data directory from command line
+data_dir = os.path.abspath(sys.argv[1])+'/'
+
+
+# Classify all raw data (red and blue arm)
+obs_metadatas = classify(data_dir)
+
+
+for arm in obs_metadatas.keys():
+
+    obs_metadata = obs_metadatas[arm]
+    # Get the project directory from the file location 
+    proj_dir = os.path.dirname(__file__)
+    
+    # Check observing mode
+    sci_im = obs_metadatas[arm]['sci'][0]['sci'][0]+'.fits'
+    if pywifes.is_nodshuffle(data_dir+sci_im):
+        obs_mode = 'ns'
+    else:
+        obs_mode = 'class'    
+
+    #************************************************************************
+    #*****       LOAD JSON FILE WITH USER REDUCTION SETUP              *****
+    #************************************************************************
+    # Read the JSON file
+    file_path = 'params_'+obs_mode+'.json'
+    with open(file_path, 'r') as f:
+        proc_steps = json.load(f)
+    #************************************************************************
+
+
+    # Check if the reduc_blue/red folders already exists and create them if required
+    out_dir = os.path.join(proj_dir, 'reduc_'+arm) 
+    if not os.path.exists(out_dir):
+        os.mkdir(out_dir)
+    else:
+        print(f"Folder '{'reduc_'+arm}' already exists.")
+
+    
+    calib_prefix = os.path.join(out_dir, 'wifesR_20150314')
+
+    # Some WiFeS specific things
+    my_data_hdu=0
+
+    # SET MULTITHREAD ?
+    #~ multithread=False
+    multithread=False
+
```

Comment:
This multithread is unused. my_data_hdu and  skip_done are, but honestly they Shouldn't be. We can update that after this pull request, but let's remove multithread now.

Reply:
Yes, agree. Done.